specs/1137-CLNT-client_api.md

```diff
@@ -0,0 +1,45 @@
+# Client API
+
+## Methods
+
+### Connect (client.connect_wallet)
+
+- If I connect and it is not approved then I get an error telling me the user rejected the connection
+- If I connect and it is approved then I do not get an error
+- If I connect and it is approved and I try to connect again then the connection is automatically approved
+- If I connect without a chain Id I am connected to the default network for that extension
+- If I connect with a non-existent chain Id I get an error telling me the network was not found
+- If I connect with a chainId I am connected to that chain
+- If I connect with a chainId and then connect again with a different chainId then I get an error indicating that I am already connected to a different chain
+
+### Get Chain Id (client.get_chain_id)
+
+- If I am not connected then I receive the selected chainId for the wallet
+- If I connect and then get the chain id then I get the chain id of that network
+
+### Is Connected (client.is_connected)
```

Comment:
Maybe excessive, but what if a user doesn't approve or reject the connection?

Reply:
Worth adding I shall do so 👍